File: desktop/Protocol/EventTypes.cs
namespace Jodu.Desktop.Protocol;

public static class EventTypes
{
    public const string ClipboardUpdate = "CLIPBOARD_UPDATE";
    public const string Telemetry = "TELEMETRY";
    public const string OtpDetected = "OTP_DETECTED";
    public const string MediaControl = "MEDIA_CONTROL";
    public const string MediaState = "MEDIA_STATE";
    public const string PingDevice = "PING_DEVICE";
    public const string Discovery = "DISCOVERY";
    public const string PairRequest = "PAIR_REQUEST";
    public const string PairResponse = "PAIR_RESPONSE";
}

public static class MediaActions
{
    public const string Play = "PLAY";
    public const string Pause = "PAUSE";
    public const string Next = "NEXT";
    public const string Previous = "PREVIOUS";
    public const string VolumeUp = "VOLUME_UP";
    public const string VolumeDown = "VOLUME_DOWN";
}

public static class JoduPorts
{
    public const int Discovery = 19283;
    public const int WebSocket = 19284;
    public const int FileHttp = 19285;
}

File: android/app/src/main/java/com/jodu/app/protocol/EventTypes.kt
package com.jodu.app.protocol

object EventTypes {
    const val CLIPBOARD_UPDATE = "CLIPBOARD_UPDATE"
    const val TELEMETRY = "TELEMETRY"
    const val OTP_DETECTED = "OTP_DETECTED"
    const val MEDIA_CONTROL = "MEDIA_CONTROL"
    const val MEDIA_STATE = "MEDIA_STATE"
    const val PING_DEVICE = "PING_DEVICE"
    const val DISCOVERY = "DISCOVERY"
    const val PAIR_REQUEST = "PAIR_REQUEST"
    const val PAIR_RESPONSE = "PAIR_RESPONSE"
}

object MediaActions {
    const val PLAY = "PLAY"
    const val PAUSE = "PAUSE"
    const val NEXT = "NEXT"
    const val PREVIOUS = "PREVIOUS"
    const val VOLUME_UP = "VOLUME_UP"
    const val VOLUME_DOWN = "VOLUME_DOWN"
}

object JoduPorts {
    const val DISCOVERY = 19283
    const val WEB_SOCKET = 19284
    const val FILE_HTTP = 19286
}
